app/Models/Mship/Role.php

```diff
@@ -128,7 +128,21 @@ public function hasPermission($permission)
         return !$this->permissions->filter(function ($perm) use ($permission) {
             $stripped = preg_replace("/[^A-Za-z0-9\/]/i", '', $perm->name);
 
-            return strcasecmp($perm->name, $permission) == 0 || strcasecmp($stripped, $permission) == 0 || $perm->name == '*';
+            if (strcasecmp($perm->name, $permission) == 0 || strcasecmp($stripped, $permission) == 0 || $perm->name == '*') {
+                return true;
+            }
+
+            // Secondary fallback - Mainly for non-numeric slugs
+            // Add slashes
+            $perm_has = str_replace('/', '\/', $perm->name);
```

Comment:
Can this be done with `preg_quote()`, to escape all regex characters? https://secure.php.net/manual/en/function.preg-quote.php

Reply:
According to the docs `Note that / is not a special regular expression character.`. Did test it as well, but it didn't modify the string

Reply:
The `$delimiter` parameter can be used for that purpose
>If the optional delimiter is specified, it will also be escaped. This is useful for escaping the delimiter that is required by the PCRE functions. The / is the most commonly used delimiter.

Reply:
Must of missed that, updated